### PR TITLE
Pass params directly from EventListener to TriggerTemplate

### PR DESCRIPTION
**Pull request by ncskier**
Pass params directly from the EventListener to the TriggerTemplate instead of going though the TriggerBinding. This way, the EventListener params will be merged with the TriggerBinding params; an error is reporting if there are multiple params with the same name.

This simplifies the TriggerBinding, so we no longer need an additional interpolation variable in it. This also leads the way to potentially making the TriggerBinding optional if a user does not care about information in the event.

This also addresses #126 

# Changes
New EventListener yaml (from `examples/`):
```yaml
apiVersion: tekton.dev/v1alpha1
kind: EventListener
metadata:
  name: listener
spec:
  serviceAccountName: tekton-triggers-example-sa
  triggers:
    - binding:
        name: pipeline-binding
      template:
        name: pipeline-template
      params:
        - name: message
          value: Hello from the Triggers EventListener!
```
New TriggerBinding yaml (from `examples/`):
```yaml
apiVersion: tekton.dev/v1alpha1
kind: TriggerBinding
metadata:
  name: pipeline-binding
spec:
  params:
    - name: gitrevision
      value: $(event.head_commit.id)
    - name: gitrepositoryurl
      value: $(event.repository.url)
```

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: docs/eventlisteners.md [hunk 1]
```diff
@@ -25,13 +25,13 @@ spec:
...
         name: pipeline-template
+      params:
+        - name: message
+          value: Hello from the Triggers EventListener!
 ```
 
 ## Parameters
-`EventListener`s can provide `params` which are passed to the `TriggerBinding`
-specified by the `binding.name`. Each parameter has a `name` and a `value`.
+`EventListener`s can provide `params` which are passed to the `TriggerTemplate`
```
Comment (vtereso) [inline docs/eventlisteners.md]: This is slightly confusing? The params are passed to the `TriggerTemplate` as specified, but it also sort of sounds like the params should be under the template. Maybe just me?

File: pkg/template/resource.go [hunk 2]
```diff
@@ -117,3 +97,31 @@ func Uid() string {
 func ApplyUIDToResourceTemplate(rt json.RawMessage, uid string) json.RawMessage {
 	return bytes.Replace(rt, uidMatch, []byte(uid), -1)
 }
+
+// MergeParams merges two param arrays. An error is returned if there are
+// multiple params with the same name.
+func MergeParams(params1 []pipelinev1.Param, params2 []pipelinev1.Param) ([]pipelinev1.Param, error) {
```
Comment (vtereso) [inline pkg/template/resource.go]: IIUC, this is the same as `MergeInDefaultParams()`, but does not allow for duplicates. These could be combined (maybe), but I think they are distinct enough as well.

Comment (ncskier) [reply pkg/template/resource.go]: `MergeParams` merges two lists of `Param`s, whereas `MergeInDefaultParams` takes one list of `Param`s, and one list of `ParamSpec`s, then adds the `default` values for params not provided in the list

File: docs/eventlisteners.md (hunk 1 above)
Comment (dibyom) [reply docs/eventlisteners.md]: The resource templates declare the parameters they need while the params here specify the values for those declared params. Usually, we'd pass those values in from the event via the binding or via default values.

File: pkg/template/resource.go (hunk 2 above)
Comment (ncskier) [reply pkg/template/resource.go]: Do you think there's a more descriptive name that I can use to make this distinction more clear?

File: docs/eventlisteners.md (hunk 1 above)
Comment (ncskier) [reply docs/eventlisteners.md]: I rephrased it to hopefully make this more clear.

**Pull request merged**